docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,270 @@
+---
+title: Configuration Management
+tags: [Reference, Device Profile, Firmware Management, Software Management, Configuration Management]
+sidebar_position: 6
+description: Device profile API proposal
+---
+
+# Device profile
+
+A device profile defines any desired combination of a firmware, software and associated configurations to be installed on a device.
+Device profiles are used to get a fleet of devices into a consistent and homogeneous state by having the same set of firmware,
+software and configurations installed on all of them.
+
+The `tedge-agent` handles `device_profile` operations as follows:
+
+* Declares `device_profile` support by sending the capability message to `<root>/<device-topic-id>/cmd/device_profile` topics
+* Subscribes to `<root>/<device-topic-id>/cmd/device_profile/+` to receive `device_profile` commands.
+* The `device_profile` command payload is an ordered list of firmware, software and configuration update operations.
+* The agent processes each operation one by one, triggering sub-operations for the respective operation.
+* On successful installation of all the modules, the applied profile information is published to the same capability topic.
+* No rollback is performed on partial failures unless the subcommand for the failed module can rollback that single module.
+
+# Why device profile FAQ
+
+**Q1: Why do we need device profile when firmware, software and configuration management already exists**
+
+Performing and managing these operations individually for a long list of software and configuration items would be cumbersome,
+especially when performed on a large fleet of devices as each operation will have to be monitored and managed separately.
+Grouping them together into a single operation reduces that overhead considerably as you just have one operation to monitor
+instead of multiple.
+
+**Q2: Why not model each device profile update as a firmware update that includes the desired software and configurations as well**
+
+Although this is a more robust approach, it is not feasible on all kinds of devices,
+especially the ones that does not support delta firmware updates.
+On such devices, pushing the entire firmware binary for each iterative change would considerably increase the binary size overhead.
+
+# Requirements
+
+* Ability to control the order of execution of the operations defined in the command input
+* Ability to apply 
+
+# Device profile capability
+
+A device that supports device profile operation must declare that capability by publishing the following message:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile' '{
+  "name": "prod-profile",
+  "version: "v1"
```

Comment:
Why is this firmware version published as a capability? It rather sounds to be related to a device state such as `twin/firmware`

Reply:
Yes, the firmware version is published to the `twin/firmware` topic, below is an example:

```
[te/device/main///twin/firmware] {"name":"tedge_rugpi_tryboot","version":"20240410.1917","updated":"Tue  9 Jul 20:15:19 BST 2024"}
```

Reply:
Updated as suggested, keeping it consistent with the `firmware` twin data.